ejb3/src/main/java/org/jboss/as/ejb3/iiop/EjbCorbaServant.java

```diff
@@ -314,21 +324,59 @@ public OutputStream _invoke(final String opName, final InputStream in, final Res
                                 retVal = false;
                             }
                         } else {
-
-                            if (sc != null) {
-                                setSecurityContextOnAssociation(sc);
+                            if (this.securityDomain != null) {
+                                // an elytron security domain is available: authenticate and authorize the client before
+                                // invoking the component.
+                                ServerAuthenticationContext context = this.securityDomain.createNewAuthenticationContext();
+                                if (credential == sasCurrent) {
+                                    // TODO change this to use the SASCurrent as the credential when we have a realm that can act on it.
```

Comment:
Also there is the IdentityToken which could possibly be used to determine if AnonymousPrincipal should be used (honestly though if there's no principal name, that seems like anonymous by default to me).

Last thought on credential, the IdentityToken contains a certificate_chain property which should be pretty easy to decode into a X.500 certificate public credential, and in the absence of a name principal, and X500Principal could maybe be extracted from the resultant certificate.  Also IdentityToken contains a dn property which could maybe be direct-converted to an X500Principal.